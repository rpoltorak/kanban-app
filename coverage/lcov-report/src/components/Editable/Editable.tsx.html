<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Editable/Editable.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Editable</a> Editable.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">58.7% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>27/46</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>15/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.86% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">64.86% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>24/37</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108</td><td class="line-coverage quiet"><span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react";
&nbsp;
import "./Editable.scss";
&nbsp;
interface EditableProps {
  value: string;
  placeholder?: string;
  autoFocus?: boolean;
  minLength?: number;
  maxLength?: number;
  onChange: (value: string) =&gt; void;
}
&nbsp;
interface EditableState {
  value: string;
  isEdited: boolean;
  isError: boolean;
}
&nbsp;
export default class Editable extends React.PureComponent&lt;EditableProps, EditableState&gt; {
  public static defaultProps = {
    value: "",
  };
&nbsp;
  public state = {
    value: this.props.value,
    isEdited: !this.props.value,
    isError: false,
  };
&nbsp;
  public ref = React.createRef&lt;HTMLDivElement&gt;();
&nbsp;
  public componentDidMount = () =&gt; {
    if (this.ref.current) {
      this.ref.current.innerHTML = this.state.value;
      this.checkErrors();
      <span class="missing-if-branch" title="else path not taken" >E</span>if (this.props.autoFocus) {
        this.ref.current.focus();
      }
    }
  }
&nbsp;
  public toggleEdit = <span class="fstat-no" title="function not covered" >() =&gt; <span class="cstat-no" title="statement not covered" ></span>this.setState(<span class="fstat-no" title="function not covered" >(</span>state: EditableState) =&gt; <span class="cstat-no" title="statement not covered" >({ isEdited: !state.isEdited }))</span>;</span>
&nbsp;
  public save = <span class="fstat-no" title="function not covered" >() =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    this.toggleEdit();</span>
<span class="cstat-no" title="statement not covered" >    this.props.onChange(this.state.value);</span>
  }
&nbsp;
  public reset = <span class="fstat-no" title="function not covered" >() =&gt; <span class="cstat-no" title="statement not covered" ></span>this.setState(<span class="fstat-no" title="function not covered" >(</span>state: EditableState) =&gt; <span class="cstat-no" title="statement not covered" >({ value: this.props.value }))</span>;</span>
&nbsp;
  public handleBlur = <span class="fstat-no" title="function not covered" >async (</span>event: React.FocusEvent&lt;HTMLDivElement&gt;) =&gt; {
<span class="cstat-no" title="statement not covered" >    event.persist();</span>
&nbsp;
<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" >    await this.checkErrors();</span></span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    if (!this.ref.current) {</span>
<span class="cstat-no" title="statement not covered" >      return;</span>
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    if (this.state.isError) {</span>
<span class="cstat-no" title="statement not covered" >      event.preventDefault();</span>
<span class="cstat-no" title="statement not covered" >      this.ref.current.focus();</span>
    } else {
<span class="cstat-no" title="statement not covered" >      this.props.onChange(this.ref.current.innerHTML);</span>
<span class="cstat-no" title="statement not covered" >      this.save();</span>
    }
  }
&nbsp;
  public checkErrors = () =&gt; {
    this.setState((state: EditableState) =&gt; ({
      isError: (!!this.props.minLength &amp;&amp; state.value.length &lt; this.props.minLength) ||
        (!!this.props.maxLength &amp;&amp; state.value.length &gt;= this.props.maxLength),
    }));
  }
&nbsp;
  public handleChange = <span class="fstat-no" title="function not covered" >(</span>event: React.FormEvent&lt;HTMLDivElement&gt;) =&gt; {
    const { <span class="cstat-no" title="statement not covered" >innerText </span>} = event.target as HTMLElement;
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.setState(<span class="fstat-no" title="function not covered" >(</span>state: EditableState) =&gt; <span class="cstat-no" title="statement not covered" >({ value: innerText }),</span> this.checkErrors);</span>
  }
&nbsp;
  public renderCounter = () =&gt; {
    return this.props.maxLength &amp;&amp; this.state.isEdited &amp;&amp; (
      &lt;div className="editable__counter"&gt;
        {this.state.value.length}/{this.props.maxLength}
      &lt;/div&gt;
    );
  }
&nbsp;
  public render() {
    return (
      &lt;div className={this.state.isError ? "editable__wrapper error" : "editable__wrapper"}&gt;
        &lt;div
          contentEditable={true}
          className="editable__input"
          ref={this.ref}
          placeholder={this.props.placeholder}
          defaultValue={this.state.value}
          onInput={this.handleChange}
          onBlur={this.handleBlur}
        /&gt;
        {this.renderCounter()}
      &lt;/div&gt;
    );
  }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 23:33:21 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
