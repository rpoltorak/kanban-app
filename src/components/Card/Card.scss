.card {
  position: relative;
  background-color: #fff;
  box-shadow: 0 1px 0 rgba(9,45,66,.25);
  border-radius: 3px;
  margin-top: 10px;
  padding: 6px 8px 2px;
  cursor: pointer;

  &__title {
    font-weight: bold;
    padding-bottom: 7px;
    margin-bottom: 7px;
    border-bottom: 1px solid #eaeaea;
  }
}

