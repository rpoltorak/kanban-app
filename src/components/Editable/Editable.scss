.editable {
  &__input {
    font-size: 14px;
    line-height: 1.5;
    min-height: 20px;
    max-width: 100%;
    resize: none;
    padding: 2px 5px;

    &:focus {
      outline: 0;
      border-bottom: 1px solid #000;
    }

    &:empty:before {
      content:attr(placeholder);
      color: #aaa;
    }
  }

  &__wrapper {
    margin-bottom: 5px;
  }

  &__counter {
    font-size: 12px;
    text-align: right;
  }
}

.error {
  color: red;

  .editable {
    &__input {
      border-color: red;
    }
  }
}


