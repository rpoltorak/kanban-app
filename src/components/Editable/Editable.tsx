import React from "react";

import "./Editable.scss";

interface EditableProps {
  value: string;
  placeholder?: string;
  autoFocus?: boolean;
  minLength?: number;
  maxLength?: number;
  onChange: (value: string) => void;
}

interface EditableState {
  value: string;
  isEdited: boolean;
  isError: boolean;
}

export default class Editable extends React.PureComponent<EditableProps, EditableState> {
  public static defaultProps = {
    value: "",
  };

  public state = {
    value: this.props.value,
    isEdited: !this.props.value,
    isError: false,
  };

  public ref = React.createRef<HTMLDivElement>();

  public componentDidMount = () => {
    if (this.ref.current) {
      this.ref.current.innerHTML = this.state.value;
      this.checkErrors();
      if (this.props.autoFocus) {
        this.ref.current.focus();
      }
    }
  }

  public toggleEdit = () => this.setState((state: EditableState) => ({ isEdited: !state.isEdited }));

  public save = () => {
    this.toggleEdit();
    this.props.onChange(this.state.value);
  }

  public reset = () => this.setState((state: EditableState) => ({ value: this.props.value }));

  public handleBlur = async (event: React.FocusEvent<HTMLDivElement>) => {
    event.persist();

    await this.checkErrors();

    if (!this.ref.current) {
      return;
    }

    if (this.state.isError) {
      event.preventDefault();
      this.ref.current.focus();
    } else {
      this.props.onChange(this.ref.current.innerHTML);
      this.save();
    }
  }

  public checkErrors = () => {
    this.setState((state: EditableState) => ({
      isError: (!!this.props.minLength && state.value.length < this.props.minLength) ||
        (!!this.props.maxLength && state.value.length >= this.props.maxLength),
    }));
  }

  public handleChange = (event: React.FormEvent<HTMLDivElement>) => {
    const { innerText } = event.target as HTMLElement;

    this.setState((state: EditableState) => ({ value: innerText }), this.checkErrors);
  }

  public renderCounter = () => {
    return this.props.maxLength && this.state.isEdited && (
      <div className="editable__counter">
        {this.state.value.length}/{this.props.maxLength}
      </div>
    );
  }

  public render() {
    return (
      <div className={this.state.isError ? "editable__wrapper error" : "editable__wrapper"}>
        <div
          contentEditable={true}
          className="editable__input"
          ref={this.ref}
          placeholder={this.props.placeholder}
          defaultValue={this.state.value}
          onInput={this.handleChange}
          onBlur={this.handleBlur}
        />
        {this.renderCounter()}
      </div>
    );
  }
}
