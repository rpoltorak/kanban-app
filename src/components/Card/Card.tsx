import React from "react";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch, AnyAction } from "redux";

import { deleteCard, updateCard, DeleteCardAction, UpdateCardAction } from "../../actions/cardActions";
import { RootState } from "../../reducers/rootReducer";
import { getCardById } from "../../selectors/cardsSelector";
import Editable from "../Editable";

import "./Card.scss";

interface CardOwnProps {
  id: string;
  laneId: string;
}

interface CardStateProps {
  title: string;
  description: string;
}

interface CardDispatchProps {
  deleteCard: () => DeleteCardAction;
  updateTitle: (title: string) => UpdateCardAction;
  updateDescription: (description: string) => UpdateCardAction;
}

type CardProps = CardOwnProps & CardStateProps & CardDispatchProps;

export const Card: React.SFC<CardProps> = (props: CardProps) => (
  <div className="card">
    <div className="card__title">
      <Editable
        value={props.title}
        onChange={props.updateTitle}
        minLength={1}
      ></Editable>
    </div>
    <div className="card__description">
      <Editable
        value={props.description}
        onChange={props.updateDescription}
        placeholder="description"
      ></Editable>
    </div>
    <button className="btn--delete" onClick={props.deleteCard}>&times;</button>
  </div>
);

const mapStateToProps = (state: RootState, ownProps: CardOwnProps): CardStateProps => {
  const { title, description } = getCardById(state, ownProps.id);

  return { title, description };
};

const mapDispatchToProps = (dispatch: Dispatch<AnyAction>, ownProps: CardOwnProps): CardDispatchProps => (
  bindActionCreators({
    deleteCard: () => deleteCard(ownProps.id, ownProps.laneId),
    updateTitle: (title: string) => updateCard(ownProps.id, { title }),
    updateDescription: (description: string) => updateCard(ownProps.id, { description }),
  }, dispatch)
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Card);
