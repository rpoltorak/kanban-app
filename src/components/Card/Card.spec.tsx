import React from "react";
import { shallow } from "enzyme";

import { Card } from "./Card";
import Editable from "../Editable";

describe("Card Component", () => {
  test("to match snapshot", () => {
    const card = shallow(
      <Card
        id={"21"}
        laneId={"1"}
        title="some title"
        description="some description"
        deleteCard={jest.fn()}
        updateTitle={jest.fn()}
        updateDescription={jest.fn()}
      />,
    );

    expect(card).toMatchSnapshot();
  });

  test("should render correctly when passed props", () => {
    const card = shallow(
      <Card
        id={"21"}
        laneId={"1"}
        title="some title"
        description="some description"
        deleteCard={jest.fn()}
        updateTitle={jest.fn()}
        updateDescription={jest.fn()}
      />,
    );

    expect(card.find(".card").exists()).toBe(true);
    expect(card.find(".btn--delete").exists()).toBe(true);
    expect(card.find(Editable).length).toBe(2);
  });

  test("should call deleteCard function after click on delete button", () => {
    const spyOnDeleteCard = jest.fn();
    const card = shallow(
      <Card
        id={"21"}
        laneId={"1"}
        title="some title"
        description="some description"
        deleteCard={spyOnDeleteCard}
        updateTitle={jest.fn()}
        updateDescription={jest.fn()}
      />,
    );

    card.find(".btn--delete").simulate("click");

    expect(spyOnDeleteCard).toHaveBeenCalled();
  });
});
