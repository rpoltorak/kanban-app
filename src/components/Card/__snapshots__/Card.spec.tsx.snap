// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    deleteCard={[MockFunction]}
    description="some description"
    id="21"
    laneId="1"
    title="some title"
    updateDescription={[MockFunction]}
    updateTitle={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card__title"
        >
          <Editable
            minLength={1}
            onChange={[MockFunction]}
            value="some title"
          />
        </div>,
        <div
          className="card__description"
        >
          <Editable
            onChange={[MockFunction]}
            placeholder="description"
            value="some description"
          />
        </div>,
        <button
          className="btn--delete"
          onClick={[MockFunction]}
        >
          ×
        </button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Editable
            minLength={1}
            onChange={[MockFunction]}
            value="some title"
          />,
          "className": "card__title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "minLength": 1,
            "onChange": [MockFunction],
            "value": "some title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Editable
            onChange={[MockFunction]}
            placeholder="description"
            value="some description"
          />,
          "className": "card__description",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [MockFunction],
            "placeholder": "description",
            "value": "some description",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "×",
          "className": "btn--delete",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "×",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card__title"
          >
            <Editable
              minLength={1}
              onChange={[MockFunction]}
              value="some title"
            />
          </div>,
          <div
            className="card__description"
          >
            <Editable
              onChange={[MockFunction]}
              placeholder="description"
              value="some description"
            />
          </div>,
          <button
            className="btn--delete"
            onClick={[MockFunction]}
          >
            ×
          </button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Editable
              minLength={1}
              onChange={[MockFunction]}
              value="some title"
            />,
            "className": "card__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "minLength": 1,
              "onChange": [MockFunction],
              "value": "some title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Editable
              onChange={[MockFunction]}
              placeholder="description"
              value="some description"
            />,
            "className": "card__description",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [MockFunction],
              "placeholder": "description",
              "value": "some description",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "×",
            "className": "btn--delete",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "×",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
